File: golang-devildyw/simple/L226翻转二叉树.go
package main

func invertTree(root *TreeNode) *TreeNode {

	if root == nil {
		return nil
	}

	// 自顶向下 遍历到一层时 先把它的左右子节点交换
	root.Left, root.Right = root.Right, root.Left
	// 再向下
	root.Left = invertTree(root.Left)
	root.Right = invertTree(root.Right)
	return root
}

File: java-devildyw/src/main/java/top/dyw/leetcode/Hot100/L226翻转二叉树.java
package top.dyw.leetcode.Hot100;

public class L226翻转二叉树 {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }

        TreeNode right = invertTree(root.left);
        TreeNode left = invertTree(root.right);

        root.left = left;
        root.right = right;

        return root;
    }
}
